src/main/java/com/bakdata/common_kafka_streams/util/ErrorUtil.java

```diff
@@ -14,47 +38,82 @@
 import org.apache.avro.io.JsonEncoder;
 import org.apache.avro.specific.SpecificDatumWriter;
 import org.apache.avro.specific.SpecificRecord;
+import org.apache.commons.compress.utils.Charsets;
 import org.apache.kafka.common.errors.SerializationException;
 
+/**
+ * This class provides utility methods for dealing with errors in Kafka streams, such as serializing values to string
+ * and classifying errors as revoverable.
+ */
 @Slf4j
 @UtilityClass
 public class ErrorUtil {
 
-    private static String writeAsJson(final SpecificRecord itemRecord) throws IOException {
-        final Schema targetSchema = itemRecord.getSchema();
-        final DatumWriter<SpecificRecord> writer = new SpecificDatumWriter<>(targetSchema);
-        try (final ByteArrayOutputStream out = new ByteArrayOutputStream()) {
-            writeJson(itemRecord, writer, out);
-            return new String(out.toByteArray());
-        }
+    /**
+     * Check if an excption is recoverable and thus shoudl be thrown so that the process is restarted by the execution
```

Comment:
typos: 
Check if an exception is recoverable and thus should be thrown, so that the process is restarted by the execution